dds/idl/dds_generator.h

```diff
@@ -846,22 +916,32 @@ class Fields {
     typedef AST_Field*& reference;
     typedef std::input_iterator_tag iterator_category;
 
-    explicit Iterator(AST_Structure* node = 0, unsigned pos = 0)
+    explicit Iterator(AST_Structure* node = 0, unsigned pos = 0, bool just_keys = false)
     : node_(node)
     , pos_(pos)
+    , just_keys_(just_keys)
     {
-      check();
+      validate_pos();
     }
 
-    bool valid() const {
+    bool valid() const
+    {
       return node_ && pos_ < node_->nfields();
     }
 
-    void check()
+    bool check()
     {
       if (!valid()) {
         node_ = 0;
         pos_ = 0;
+        return false;
+      }
+      return true;
+    }
+
+    void validate_pos()
+    {
+      for (; check() && just_keys_ && !be_global->is_key(**this); ++pos_) {
       }
```

Comment:
Just put this on one line

Reply:
I could, but I think that goes against the spirit of the guidelines.

Reply:
Okay, just checked and I agree, I will resolve.